web_src/js/modules/observer.ts

```diff
@@ -0,0 +1,89 @@
+import {isDocumentFragmentOrElementNode} from '../utils/dom.ts';
+
+type DirElement = HTMLInputElement | HTMLTextAreaElement;
```

Comment:
Ah, that's supposed to be `direction`,  not `directory`.
I was confused for a second.

Reply:
That's from old "dirauto.js"

This file is a rewritten of "dirauto.js", more than a half of code is from old code.

![image](https://github.com/user-attachments/assets/de6ee09e-75ac-4540-9313-2375f72cbc9f)